nixos/modules/services/databases/influxdb2.nix

```diff
@@ -76,17 +179,251 @@ in
             description = "API Token to set for the admin user. Don't use a file from the nix store!";
           };
         };
+
+        organizations = mkOption {
+          description = "Organizations to provision.";
+          example = literalExpression ''
+            {
+              myorg = {
+                description = "My organization";
+                buckets.mybucket = {
+                  description = "My bucket";
+                  retention = 31536000; # 1 year
+                };
+                auths.mytoken = {
+                  readBuckets = ["mybucket"];
+                  tokenFile = "/run/secrets/mytoken";
+                };
+              };
+            }
+          '';
+          default = {};
+          type = types.attrsOf (types.submodule (organizationSubmod: let
```

Comment:
Also we may move these submodule declarations to the let/in block above, making the module less cluttered.